### ISSUE TITLE
ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast#7049]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:103)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:835)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:849)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:97)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:137)
```

It could be related to recent changes in the mock network? #7036

**Comment by jerrinot** [comment]
It appears to be reproducible with sufficient number of repetitions.

**Comment by metanet** [comment]
is this on master?

**Comment by jerrinot** [comment]
it is

**Comment by metanet** [comment]
thanks, i will check it

**Comment by jerrinot** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/17398/

**Comment by metanet** [comment]
Closed by https://github.com/hazelcast/hazelcast/commit/4f431e04f80b92efd0a7ae082e62e39875d5c5a4

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:104)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:737)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:751)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:98)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:139)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance/com.hazelcast$hazelcast-client-new/1124/testReport/junit/com.hazelcast.client.map/ClientMapPartitionLostListenerTest/test_mapPartitionLostListener_invoked_fromOtherNode/

**Comment by cangencer** [comment]
created a build on Jenkins as I was unable to reproduce it locally: https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-issue-7049/

**Comment by cangencer** [comment]
Build failure: 

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-issue-7049/8/

```
Stacktrace

java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:105)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:737)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:751)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:99)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:140)
```

**Comment by metanet** [comment]
fixed by https://github.com/hazelcast/hazelcast/commit/e0f6f89dd1368de0cccf3df10ae8f6df517bc8b9 and https://github.com/hazelcast/hazelcast/commit/d1f8eec19bb6ee3064f106c8d6d019da5349ebb1

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:104)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:737)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:751)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:98)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:139)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-IbmJDK1.8/com.hazelcast$hazelcast-client/26/testReport/junit/com.hazelcast.client.map/ClientMapPartitionLostListenerTest/test_mapPartitionLostListener_invoked_fromOtherNode/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-IbmJDK1.8/com.hazelcast$hazelcast-client/26/testReport/junit/com.hazelcast.client.map/ClientMapPartitionLostListenerTest/test_mapPartitionLostListener_invoked_fromOtherNode/

**Comment by ihsandemir** [comment]
After some review and analysis, we spotted that the cause was too early event firing before the map container is constructed at the backup node. The related log is at:
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-issue-7049/23/artifact/hazelcast-client/target/surefire-reports/com.hazelcast.client.map.ClientMapPartitionLostListenerTest-output.txt

We observe the following log during the test run:

 8900 INFO: Publishing InternalPartitionLostEvent
 8901 15:30:01,457  INFO [onPartitionLost] test_mapPartitionLostListener_invoked_fromOtherNode - Number of entries:0 , entries:[]
 8902 Jan 05, 2016 3:30:01 PM com.hazelcast.client.map.ClientMapPartitionLostListenerTest test_mapPartitionLostListener_invoked_fromOtherNode
 8903 INFO: Published InternalPartitionLostEvent

Since no container exist, no event is delivered and that was the reason for failure.

We decided with Basri to fix this problem in the test code by guaranteeing the creation of the map container at the backup before firing the event.

**Comment by ihsandemir** [comment]
PR #7254 fixed this issue.

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest$1.run(ClientMapPartitionLostListenerTest.java:108)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:851)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:865)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.assertMapPartitionLostEventEventually(ClientMapPartitionLostListenerTest.java:102)
	at com.hazelcast.client.map.ClientMapPartitionLostListenerTest.test_mapPartitionLostListener_invoked_fromOtherNode(ClientMapPartitionLostListenerTest.java:145)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-IbmJDK1.7/com.hazelcast$hazelcast-client-legacy/391/testReport/junit/com.hazelcast.client.map/ClientMapPartitionLostListenerTest/test_mapPartitionLostListener_invoked_fromOtherNode/

**Comment by jerrinot** [comment]
fixed by #8293
